vscode/src/non-stop/FixupTypingUI.ts

```diff
@@ -1,34 +1,120 @@
+    if (lastIndex === -1) {
+        // Return the original string if "@" is not found
+        return str
+    }
+    return str.slice(0, lastIndex)
+}
+
+function getLabelForContextFile(file: ContextFile): string {
+    const isFileType = file.type === 'file'
+    const rangeLabel = file.range ? `:${file.range?.start.line}-${file.range?.end.line}` : ''
+    if (isFileType) {
+        return `${file.path?.relative}${rangeLabel}`
+    }
+    return `${file.path?.relative}${rangeLabel}#${file.fileName}`
+}
+
+/* Match strings that end with a '@' followed by any characters except a space */
+const MATCHING_CONTEXT_FILE_REGEX = /@(\S+)$/
+
+/* Match strings that end with a '@#' followed by any characters except a space */
+const MATCHING_SYMBOL_REGEX = /@#(\S+)$/
+
+const MAX_FUZZY_RESULTS = 20
+const FILE_HELP_LABEL = 'Search for a file to include, or type # to search symbols..'
+const SYMBOL_HELP_LABEL = 'Search for a symbol to include...'
+const NO_MATCHES_LABEL = 'No matches found'
+
+interface FixupMatchingContext {
+    /* Unique identifier for the context, shown in the input value but not necessarily in the quick pick selector */
+    key: string
+    /* If present, will override the key shown in the quick pick selector */
+    shortLabel?: string
+    file: ContextFile
+}
+
+interface QuickPickParams {
+    title?: string
+    placeholder?: string
+    initialValue?: string
+    initialSelectedContextFiles?: ContextFile[]
+    prefix?: string
+}
+
 /**
  * The UI for creating non-stop fixup tasks by typing instructions.
  */
 export class FixupTypingUI {
     constructor(private readonly taskFactory: FixupTaskFactory) {}
 
-    public async getInstructionFromQuickPick({
+    private async getMatchingContext(instruction: string): Promise<FixupMatchingContext[] | null> {
+        const symbolMatch = instruction.match(MATCHING_SYMBOL_REGEX)
+        if (symbolMatch) {
+            const symbolResults = await getSymbolContextFiles(symbolMatch[1], MAX_FUZZY_RESULTS)
+            return symbolResults.map(result => ({
+                key: getLabelForContextFile(result),
+                file: result,
+                shortLabel: `${result.kind === 'class' ? '$(symbol-structure)' : '$(symbol-method)'} ${
+                    result.fileName
+                }`,
+            }))
+        }
+
+        const fileMatch = instruction.match(MATCHING_CONTEXT_FILE_REGEX)
+        if (fileMatch) {
+            const cancellation = new vscode.CancellationTokenSource()
+            const fileResults = await getFileContextFiles(fileMatch[1], MAX_FUZZY_RESULTS, cancellation.token)
+            return fileResults.map(result => ({
+                key: getLabelForContextFile(result),
+                file: result,
+            }))
+        }
+
+        return null
+    }
+
+    public async getInputFromQuickPick({
         title = `${EDIT_COMMAND.description} (${EDIT_COMMAND.slashCommand})`,
         placeholder = 'Your instructions',
```

Comment:
```suggestion
        placeholder = 'Your instructions (@ to include additional context from file)',
```

Should we update the instruction to mention the new @ feature?

Reply:
Yes! Will update, ty